modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchViewModel.kt

```diff
@@ -91,6 +91,7 @@ class SearchViewModel @Inject constructor(
     }
 
     fun updateSearchQuery(query: String, immediate: Boolean = false) {
+        if (_state.value.searchTerm == query) return
```

Comment:
Maybe a comment would be helpful here to know what this is protecting against. I don't feel strongly, so feel free to leave as-is.
```suggestion
        // Prevent updating the search query when navigating back to the search results after tapping on a result.
        if (_state.value.searchTerm == query) return
```